src/main/java/life/qbic/io/PetabParser.java

```diff
@@ -0,0 +1,85 @@
+package life.qbic.io;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import life.qbic.model.petab.PetabMetadata;
+
+public class PetabParser {
+
+  private final String META_INFO_YAML = "metaInformation.yaml";
+
+  public PetabMetadata parse(String dataPath) {
+
+    File directory = new File(dataPath);
+    List<String> sourcePetabReferences = new ArrayList<>();
+
+    File yaml = findYaml(directory);
+    if (yaml != null) {
+      BufferedReader reader = null;
+      try {
+        reader = new BufferedReader(new FileReader(yaml));
+        boolean inIDBlock = false;
+        while (true) {
+          String line = reader.readLine();
+          if (line == null) {
+            break;
+          }
+          if(inIDBlock && line.contains(":")) {
+            inIDBlock = false;
+          }
+          if(inIDBlock) {
+            String[] tokens = line.split("-");
+            if(tokens.length == 3) {
+              String datasetCode = tokens[1].strip()+"-"+tokens[2].strip();
+              sourcePetabReferences.add(datasetCode);
+            }
+          }
+          if (line.contains("openBISSourceIds:")) {
+            inIDBlock = true;
+          }
```

Comment:
I think this section would benefit greatly from a small refactor into distinct methods that outline why the steps are taken for each if clause.  Additionally these methods can more easily specify what happens if the input is malformed (token length != 3, why is the inIdBlock set to true even though it could be previously set to false etc.)